src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -262,27 +306,53 @@ func (d *downsamplerAndWriter) WriteBatch(
 		}
 	)
 
-	if d.store != nil {
+	if d.shouldWrite(overrides) {
 		// Write unaggregated. Spin up all the background goroutines that make
 		// network requests before we do the synchronous work of writing to the
 		// downsampler.
+		overrideStoragePolicies, ok := d.writeOverrideStoragePolicies(overrides)
 		for iter.Next() {
-			wg.Add(1)
 			tags, datapoints, unit := iter.Current()
-			d.workerPool.Go(func() {
-				err := d.store.Write(ctx, &storage.WriteQuery{
-					Tags:       tags,
-					Datapoints: datapoints,
-					Unit:       unit,
-					Attributes: storage.Attributes{
-						MetricsType: storage.UnaggregatedMetricsType,
-					},
+			if !ok {
```

Comment:
could you have an if statement that determines what the `writeFn` is here, and then use the same worker pool logic and error handling for both code paths?

Reply:
Sure thing, updating.